web/models/session.ex

```diff
@@ -0,0 +1,37 @@
+defmodule ElixirBlog.Session do
+  import Ecto.Query
+
+  alias ElixirBlog.User
+  alias ElixirBlog.Repo
+
+  require Logger
+
+  def login(params) do
+    user = find_user_by_email(params["email"])
+    case authenticate(user, params["password"]) do
+      true -> {:ok, user}
+      _    -> {:error}
+    end
+  end
+
+  def current_user(conn) do
+    id = Plug.Conn.get_session(conn, :current_user) || 0
```

Comment:
I would handle the no session case other than `0`. Don't hit the Repo if you don't need to